File: Tutorials/social-media-scraping/assets/smp scraping example.R
#####################################################
#===============Connect to Twitter==================#
# Connect to Twitter API using your authentication  #
# credentials obtained from Twitter dev             #
#####################################################

library(twitteR)

consumer_key = "yourconsumerkey"
consumer_secret = "yourconsumersecretkey"
access_token = "youraccesstoken"
access_secret = "youraccesssecret"

setup_twitter_oauth(consumer_key, 
                    consumer_secret, 
                    access_token, 
                    access_secret)

tweet1 = searchTwitter(searchString = 'from:BYUfootball+stadium')

tweet1

tweet2 = searchTwitter(searchString = 'byu+football+jersey+until:2019-09-20+since:2019-09-01', n=20)

tweet2

tweet3 = searchTwitter(searchString = 'BYU+football', resultType = 'popular')

tweet3

File: Tutorials/social-media-scraping/assets/smp scraping example.py
#################################################
#=================Twitter Data==================#
# Authenticate to Twitter using tokens and keys #
# and query data                                #
#################################################

import twitter

apikey = 'yourconsumersecret'
apisecretkey = 'yourconsumersecretkey'
accesstok = 'youraccesstoken'
accesstoksec = 'youraccesstokensecret'

twitconn = twitter.Api(consumer_key=apikey,
                       consumer_secret=apisecretkey,
                       access_token_key=accesstok,
                       access_token_secret=accesstoksec)

# Verify your credentials
twitconn.VerifyCredentials()

tweet1 = twitconn.GetSearch(raw_query='q=from%3ABYUfootball%20stadium')

tweet2 = twitconn.GetUserTimeline(screen_name='BYUfootball', count=30)

tweet2

for item in tweet2:
    print(item.text)


tweet3 = twitconn.GetSearch(raw_query='q=from%3ABYUfootball%20until%3A2019-09-27%20since%3A2019-09-01')

for item in tweet3:
    print(item.id, item.text)

tweet4 = twitconn.GetSearch(raw_query='q=from%3ABYUfootball', until='2019-09-27', since='2019-09-01')


twitteruser = 'BYUfootball'
tweet5 = twitconn.GetUserTimeline(screen_name=twitteruser,
                                  count=200,
                                  include_rts=False,
                                  exclude_replies=True)